modules/internal/ChapelError.chpl

```diff
@@ -20,14 +20,176 @@
 module ChapelError {
   use ChapelStandard;
 
+  // Base class for errors
+  // TODO: should base class Error have a string message at all?
+  // TODO: should Error include list pointers for ErrorGroup?
   class Error {
     var msg: string;
 
+    proc Error() {
+      _next = nil;
+    }
+
     proc Error(_msg: string) {
       msg = _msg;
+      _next = nil;
+    }
+
+    proc writeThis(f) {
+      //f <~> "Error: " <~> msg;
+      f <~> "{msg = " <~> msg <~> "}";
+    }
+
+    pragma "no doc"
+    var _next: Error = nil; // managed by lock in record ErrorGroupRecord
```

Comment:
This is an odd placement of a field. I'd prefer for all fields to be declared together.